### PR TITLE
Create simpler units of code based on BetterCodeHub reports

### PR DESCRIPTION
**Pull request by ericcornelissen**
- Relevant Issues: -
- Related Pull Requests: #161 

* * *

## What
Improve code quality by making simpler units of code based on the BetterCodeHub report

## Why
To improve code quality.

## How
See the improvements in the BetterCodeHub results:
progress

## Alternative implementation
None considered

## Notes
n/a

### CODE REVIEW

File: app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java [hunk 1]
```diff
@@ -34,106 +34,114 @@
...
+            this.newDevices.add(device);
+            this.newDevicesArrayAdapter.add(device.getName() + "\n" + device.getAddress());
+        }
+    }
+
+    /**
+     * Set message when no devices
+     */
+    private void setNoDevicesFound() {
+        this.setTitle("Select device");
```
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java]: Can you put this in strings.xml and use getString(R.id.....)

File: app/src/main/res/layout/activity_bluetooth_connection.xml [hunk 2]
```diff
@@ -27,7 +27,7 @@
                 android:backgroundTint="@color/colorPrimary"
                 android:src="?attr/actionModeCloseDrawable"
                 android:contentDescription="@string/contact_title"
-                android:onClick="onBackButtonClick"/>
+                android:onClick="cancelBluetoothConnectionActivity"/>
```
Comment (jverbraeken) [inline app/src/main/res/layout/activity_bluetooth_connection.xml]: Maybe onCancelBluetoothConnectionActivityClick?

File: app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java (hunk 1 above)
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java]: 👍

File: app/src/main/res/layout/activity_bluetooth_connection.xml (hunk 2 above)
Comment (ericcornelissen) [inline app/src/main/res/layout/activity_bluetooth_connection.xml]: 👍

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -38,6 +38,8 @@
     <string name="edit_menu_item">Edit</string>
     <string name="delete_menu_item">Delete</string>
 
+    <string name="select_device">Select device</string>
+    <string name="scanning">Scanning...</string>
```
Comment (TheBonheurs) [inline app/src/main/res/values/strings.xml]: Why has this been added in your pr while also being part of @clenrock 's pr?

File: app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java [hunk 4]
```diff
@@ -34,106 +34,114 @@
 /**
  * This Bluetooth activity class establishes and manages a bluetooth connection.
  */
-
 @SuppressWarnings({"PMD.TooManyMethods", "PMD.NullAssignment", "PMD.EmptyWhileStmt"})
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java]: Not sure if all those suppressions hold as their description seems out of date

File: app/src/main/java/com/nervousfish/nervousfish/activities/ActivateBluetoothActivity.java [hunk 5]
```diff
@@ -37,81 +35,98 @@
...
-            new SweetAlertDialog(this, SweetAlertDialog.WARNING_TYPE)
-                    .setTitleText(getString(R.string.activating_bluetooth_went_wrong))
-                    .setContentText(getString(R.string.activating_bluetooth_went_wrong_explanation))
-                    .setConfirmText(getString(R.string.dialog_ok))
-                    .setConfirmClickListener(new FailedOnClickListener())
-                    .show();
-        } else if (resultCode == RESULT_CODE_FINISH_BLUETOOTH_ACTIVITY && requestCode == REQUEST_CODE_BLUETOOTH_ACTIVITY) {
-            finish();
-        }
+    public void onCancelActivateBluetoothActivityClick(final View v) {
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/activities/ActivateBluetoothActivity.java]: Nice refactor!

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (ericcornelissen) [inline app/src/main/res/values/strings.xml]: Not sure, they were introduced after a merge with develop...

File: app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java (hunk 4 above)
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java]: Don't know, didn't at them >.<'

**Pull request merged**